### PR TITLE
5 add models gpt 4

### PR DESCRIPTION
**Pull request by sanyakosivchuk**
to select model use choose it when initializing client with model:
also gpt-4-0314 and text-davinci-003 seems to be deprecated more here: https://platform.openai.com/docs/deprecations

example:
api_key = "YOUR API KEY"
messages = "Who is the best chess player in history?"

result = RubyAI::Client.new(api_key, messages, model: "gpt-4").call
puts result.dig("choices", 0, "message", "content")

### CODE REVIEW

**Comment by sanyakosivchuk** [comment]
i merged changes from "add faraday gem" branch into this branch to work on newer version

File: lib/rubyai.rb [hunk 1]
```diff
@@ -31,17 +40,16 @@ def call
 
     def body
       {
-        'model': model,
-        'messages': [{"role": "user", "content": messages}],
-        'temperature': temperature
+        'model': MODELS[@model],
```
Comment (alexshapalov) [inline lib/rubyai.rb:43]: - Do we have a default model now? 
- Can we add def mode:l gpt-4
- Can you add some unit test for test model versions

**Review by alexshapalov — CHANGES_REQUESTED** [review]

**Comment by alexshapalov** [comment]
Also, pls update Readme.MD

```
api_key = "YOUR API KEY"
messages = "Who is the best chess player in history?"

result = RubyAI::Client.new(api_key, messages, model: "gpt-4").call
puts result.dig("choices", 0, "message", "content")
```

File: lib/rubyai.rb (hunk 1 above)
Comment (sanyakosivchuk) [reply lib/rubyai.rb:43]: Default model is gpt 3.5 turbo, but ill add instance to choose model set by default

File: spec/client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rspec'
+require_relative '../lib/rubyai'
+
+RSpec.describe RubyAI::Client do
+  describe '#mode' do
+    it 'changes the model' do
+      client = RubyAI::Client.new('your_api_key', 'Who am i?')
```
Comment (alexshapalov) [inline spec/client_spec.rb:7]: We need to mock this data.
We can not use real api_key in the test.

Comment (sanyakosivchuk) [reply spec/client_spec.rb:7]: it works without entering the key

**Pull request merged**